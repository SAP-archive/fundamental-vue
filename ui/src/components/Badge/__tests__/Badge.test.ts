import { assert } from "chai";
import { mount, shallowMount } from "@vue/test-utils";
import { Badge } from "../";

describe("Badge", () => {
  test("renders correctly", () => {
    const successBadge = mount(Badge, {
      propsData: {
        type: "success",
        id: "successBadge",
        filled: true,
        pill: true
      }
    });
    const errorBadge = mount(Badge, { propsData: { type: "error" } });
    expect(successBadge.element).toMatchSnapshot();
    expect(errorBadge.element).toMatchSnapshot();

    expect(successBadge.classes("fd-badge--filled")).toBe(true);
    expect(successBadge.classes("fd-badge--pill")).toBe(true);
    expect(successBadge.classes("fd-badge--success")).toBe(true);
    expect(errorBadge.classes("fd-badge--error")).toBe(true);
  });
  it("renders default slot when passed", () => {
    const dummySlot = "Slot text";
    const wrapper = shallowMount(Badge, {
      slots: {
        default: dummySlot
      }
    });
    assert.include(wrapper.text(), dummySlot);
  });
});
