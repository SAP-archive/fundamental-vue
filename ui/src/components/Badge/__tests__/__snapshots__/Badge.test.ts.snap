// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Badge renders correctly 1`] = `
<span
  class="fd-badge fd-badge--filled fd-badge--pill fd-badge--success"
  id="successBadge"
/>
`;

exports[`Badge renders correctly 2`] = `
<span
  class="fd-badge fd-badge--error"
/>
`;
